File: JavaScript/e007.js
let counter = 1;
let n = 2;
while (counter < 10001) {
    n += 1;
    if (isprime(n)) {
        counter++;
    }
}
let answer = n;
console.log(answer);


function isprime(n) {

    if (n === 2 || n === 3) return true;
    if (n < 2 || n % 2 === 0) return false;
    if (n < 9) return true;
    if (n % 3 === 0) return false;

    let f = 5;
    while (f <= Math.floor(Math.sqrt(n))) {
        if (n % f === 0) return false;
        if (n % (f + 2) === 0) return false;
        f += 6;
    }
    return true;
}

File: Python/e007.py
"""
Problem 7: By listing the first six prime numbers: 2, 3, 5, 7, 11, and 13,
we can see that the 6th prime is 13.
What is the 10001st prime number?
"""
import myfunc

(counter, n) = (1, 2)  # 1st prime number is 2
while counter < 10001:
    n += 1
    if myfunc.isprime(n):
        counter += 1
print(f"10001st prime number is {n}")
